Framework/src/InfrastructureGenerator.cxx

```diff
@@ -354,8 +354,29 @@ void InfrastructureGenerator::generateCheckRunners(framework::WorkflowSpec& work
   // Build tasksOutputMap based on active tasks in the config
   for (const auto& [taskName, taskConfig] : config->getRecursive("qc.tasks")) {
     if (taskConfig.get<bool>("active", true)) {
-      o2::framework::InputSpec checkInput{ taskName, TaskRunner::createTaskDataOrigin(), TaskRunner::createTaskDataDescription(taskName) };
-      tasksOutputMap.insert({ DataSpecUtils::label(checkInput), checkInput });
+      o2::framework::InputSpec taskOutput{ taskName, TaskRunner::createTaskDataOrigin(), TaskRunner::createTaskDataDescription(taskName) };
+      tasksOutputMap.insert({ DataSpecUtils::label(taskOutput), taskOutput });
+    }
+  }
+
+  // For each external task prepare the InputSpec to be stored in tasksoutputMap
+  try {
+    for (const auto& [taskName, taskConfig] : config->getRecursive("qc.externalTasks")) {
+      (void)taskName;
+      if (taskConfig.get<bool>("active", true)) {
+        auto query = taskConfig.get<std::string>("query");
+        framework::Inputs inputs = o2::framework::DataDescriptorQueryBuilder::parse(query.c_str());
+        o2::framework::InputSpec taskOutput = inputs.at(0); // only consider the first one if several.
```

Comment:
Wait, this is actually the code which limits the number of inputs to the first one. Perhaps I though it was solved because i didn't see it in the PR which was merged before. Would it actually break anything, if any number of inputs was supported here?

Reply:
Ok, now, that is worrying, let me check, this was fixed, I am certain.

Reply:
Ok, I got it. I had fixed it in Check.cxx: https://github.com/AliceO2Group/QualityControl/pull/425/commits/61ba103edbe31716b5982ce179f7c6bd2fa11661

not in InfrastructureGenerator.cxx as this piece had been discarded (it is the piece I am adding back). I am giving it a try to actually allow any number of inputs.

Reply:
Ok, so this one is working with multiple outputs for an external task.